crates/erc20_payment_lib/src/runtime.rs

```diff
@@ -411,6 +415,17 @@ impl PaymentRuntime {
         })
     }
 
+    pub async fn get_web3_provider(&self, chain_name: &str) -> Result<Web3<Http>, PaymentError> {
```

Comment:
No `pub` function in `PaymentRuntime` should leak web3 provider details. AFAICT you're only using it for tests -- why not use `PaymentRuntime::verify_transaction` directly? This would also let you make the loose `verify_transaction` private.